solutions/level1/main.rb

```diff
@@ -1,14 +1,14 @@
 require_relative 'cart'
 require_relative 'library'
-require_relative 'book_serializer'
+require_relative 'printer'
 
 FILE = '../../requirements/level1/data.json'
 
-library = Library.new
-books_hash = JsonReaderWriter.read_file(FILE)
-library.create_books(books_hash)
+library = Library.new(FILE)
 
 cart = Cart.new
-cart.calculate_totals(library.books, [1, 2])
+cart.add_item(library.books[0])
+cart.add_item(library.books[1])
+
 receipt = cart.generate_receipt(library.books)
```

Comment:
why do you need to send all the books here ? Why does cart need to know about all the books that exist in the library ? 🤔

Reply:
Moved the books sending part onto the printer so it can 'merge' the books info along with the cart related info (total/count). And why do we send them all? In a regular scenario we would never do that, but in all the data/output samples they send all the books along with the cart info, and wanted to respect that + in higher levels we also have to display stock and stuff like that